src/watertap_contrib/seto/core/solar_energy_base.py

```diff
@@ -112,3 +121,158 @@ def calculate_scaling_factors(self):
 
         if callable(self._scaling):
             self._scaling(self)
+
+
+class SolarEnergyBase(SolarEnergyBaseData):
```

Comment:
@Matthew-Boyd is there a reason you made a separate class here? Were you having import issues?

The reason I ask is I think that this is creating another class with the same name in this file after the ``declare_process_block_data`` decorator above. I like having all the surrogate methods consolidated here but they should just be in the main class.

Reply:
I wonder if there's value in keeping them separate so that one could be used as a base for physical or surrogate models while this new one is a base just for surrogates.

Reply:
@Matthew-Boyd do you have a test for this new base class?

Reply:
But @adam-a-a does the ``delcare_process_block_data`` decorator not create a class called ``SolarEnergyBase`` using the ``SolarEnergyBaseData`` class? And thus having a class with the same name in the same file would create import issues? This is how all the models in WaterTAP work- the class is called like ``UnitModelData`` but then the decorator creates a class called ``UnitModel`` which is then what you import and what goes in the ``__init__.py`` files (see the LT-MED model as an example). 

Adding the methods in the above class should have the same effect.